File: seulgi/2024-07/P148652_유사칸토어비트열.py
def solution(n, l, r):
    answer = 0
    for l in range(l-1, r):
        if is_one(l):
            answer += 1
    return answer

def is_one(l):
    while l >= 5:
        if (l - 2) % 5 == 0:
            return False
        l //= 5

    return l != 2

File: sseni/2024-07/P148652_유사칸토어비트열.java
package com.example.algorithm;

public class P148652_유사칸토어비트열 {
    public static int solution(int n, long l, long r) {
        int answer = 0;

        for (long val = l; val <= r; val++) {
            answer += func(n, val - 1);
        }

        return answer;
    }

    static int func(int n, long l) {
        if (n == 0 || l == 0)
            return 1;
        if (l % 5 == 2)
            return 0;
        return func(n - 1, l / 5);
    }

    public static void main(String[] args) {
        System.out.println(solution(2, 4, 17));
    }
}
